bot/kodiak/dependencies.py

```diff
@@ -0,0 +1,83 @@
+import re
+from typing import List, Optional, Sequence, Tuple, TypeVar
+
+from typing_extensions import Literal
+
+title_regex = re.compile(r"from (?P<old_version>\S+) to (?P<new_version>\S+)")
+
+
+def _extract_versions(x: str) -> Optional[Tuple[str, str]]:
+    """
+    Find old and new version from PR title
+    Example:
+        title: "Bump jackson-databind from 2.9.10.1 to 2.10.0.pr1 in /LiveIngest/LiveEventWithDVR"
+        
+        result: "2.9.10.1", "2.10.0.pr1"
+    """
+    match = title_regex.search(x)
+    if match is None:
+        return None
+    group = match.groupdict()
+    if "old_version" not in group or "new_version" not in group:
+        return None
+    return group["old_version"], group["new_version"]
+
+
+# regex to split version string. Versions aren't necessarily semver.
+#
+# from dependabot: https://github.com/dependabot/dependabot-core/blob/998de3be7811956354aea077ecb180831e24012c/common/lib/dependabot/pull_request_creator/labeler.rb#L95-L115
+version_regex = re.compile(r"[.+]")
+
+
+def _parse_version_simple(x: str) -> List[str]:
+    """
+    Split version string into pieces.
+    """
+    return version_regex.split(x)
+
+
+T = TypeVar("T")
+
+
+def _get_or_none(arr: Sequence[T], index: int) -> Optional[T]:
+    try:
+        return arr[index]
+    except IndexError:
+        return None
+
+
+def _compare_versions(
+    old_version: str, new_version: str
+) -> Optional[Literal["major", "minor", "patch"]]:
+    """
+    Determine patch, like Dependabot.
+
+    https://github.com/dependabot/dependabot-core/blob/998de3be7811956354aea077ecb180831e24012c/common/lib/dependabot/pull_request_creator/labeler.rb#L92-L114
+    """
+    old_version_parts = _parse_version_simple(old_version)
+    new_version_parts = _parse_version_simple(new_version)
+
+    for part in new_version_parts[:3] + old_version_parts[:3]:
```

Comment:
why `:3`?

Reply:
Because that's what the dependabot code does and it returns the "major", "minor", "patch"-like versions if available.

```
4.1.2 => ["4","1","2"]

2018.1.2.523.b1 => ["2018","1","2"]
```